File: set-matrix-zeroes/njngwn.java
// Time Complexity: O(m*n)
// Space Complexity: O(m+n)
class Solution {
    public void setZeroes(int[][] matrix) {
        ArrayList<Integer> rowList = new ArrayList<>();
        ArrayList<Integer> colList = new ArrayList<>();

        for (int m = 0; m < matrix.length; ++m) {
            for (int n = 0; n < matrix[0].length; ++n) {
                if (matrix[m][n] != 0) continue;
                if (!rowList.contains(m)) rowList.add(m);
                if (!colList.contains(n)) colList.add(n);
            }
        }

        for (Integer row : rowList) {
            for (int n = 0;  n < matrix[0].length; ++n) {
                matrix[row][n] = 0;
            }
        }

        for (Integer col : colList) {
            for (int n = 0;  n < matrix.length; ++n) {
                matrix[n][col] = 0;
            }
        }
    }
}

File: contains-duplicate/njngwn.py
class Solution:
    # Time Complexity: O(nlogn), n: len(nums)
    # Space Complexity: O(1)
    def containsDuplicate(self, nums: List[int]) -> bool:
        nums.sort()  # O(nlogn)

        for i in range(len(nums)):
            if i > 0 and nums[i] == nums[i - 1]:
                return True

        return False
